src/auth/googleauth.ts

```diff
@@ -365,9 +357,17 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
     // For GCE, just return a default ComputeClient. It will take care of
     // the rest.
     (options as ComputeOptions).scopes = this.getAnyScopes();
-    this.cachedCredential = new Compute(options);
-    projectId = await this.getProjectIdOptional();
-    return {projectId, credential: this.cachedCredential};
+    return await this.prepareADCResponse(new Compute(options));
+  }
+
+  private async prepareADCResponse(
+    credential: JSONClient | Impersonated | Compute | T
+  ): Promise<ADCResponse> {
+    const projectId = await this.getProjectIdOptional();
+    credential.setQuotaProjectIdFromEnvironment();
```

Comment:
On second thought, this might be a bit problematic; say you explicitly set `quotaProjectId` in code, but this call constantly unsets your preference.

Reply:
For that case, how about we introduce a parameter to the adc method that provides a quota project id explicitly and we will prefer that over env var. This is the way python does.

Reply:
That seems like a user-friendly flow: if a quota id is passed explicitly, prefer it (for this one-time operation), otherwise use the credential's current quota id.